### PR TITLE
[Issue 4251][Schemas] add schema parsing verification before update

### LINKED ISSUE #4251: Schema parsing validation
**Describe the bug**
When you put a schema which has parsing issues using Avro, there is a 500 without stacktrace.

### PR DESCRIPTION
**Pull request by KannarFr**
Fixes #4251 

### Motivation

Add a way to validate new schema before try to check compatibility and add stacktrace printing on parsing schema error.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/AvroSchemaBasedCompatibilityCheck.java [hunk 1]
```diff
@@ -22,27 +22,44 @@
...
 import org.apache.avro.SchemaValidatorBuilder;
 import org.apache.pulsar.common.schema.SchemaData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * The abstract implementation of {@link SchemaCompatibilityCheck} using Avro Schema.
  */
 abstract class AvroSchemaBasedCompatibilityCheck implements SchemaCompatibilityCheck {
+    Logger log = LoggerFactory.getLogger(SchemaRegistryService.class);
```
Comment (merlimat) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/AvroSchemaBasedCompatibilityCheck.java]: `private static final Logger ...` or just add `@Slf4j` annotation to the class

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/AvroSchemaBasedCompatibilityCheck.java [hunk 2]
```diff
@@ -22,27 +22,44 @@
...
 
     @Override
-    public boolean isCompatible(SchemaData from, SchemaData to, SchemaCompatibilityStrategy strategy) {
-        Schema.Parser fromParser = new Schema.Parser();
-        Schema fromSchema = fromParser.parse(new String(from.getData(), UTF_8));
-        Schema.Parser toParser = new Schema.Parser();
-        Schema toSchema =  toParser.parse(new String(to.getData(), UTF_8));
+    public Boolean isWellFormed(SchemaData to) {
```
Comment (merlimat) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/AvroSchemaBasedCompatibilityCheck.java]: prefer `boolean` unless we'd need to return nulls

**Comment by merlimat** [comment]
rerun tests please

**Comment by merlimat** [comment]
retest this please

**Comment by KannarFr** [comment]
retest this please

**Comment by KannarFr** [comment]
retest this please

**Comment by KannarFr** [comment]
retest this please

**Comment by KannarFr** [comment]
retest this please

**Comment by merlimat** [comment]
run integration tests
run java8 tests

**Comment by merlimat** [comment]
run java8 tests

**Comment by merlimat** [comment]
run java8 tests

**Pull request merged**